### PR TITLE
UIOR-281 Save and Open PO from POL create/edit

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
To reduce the number of clicks to open POs once they have been created. If a library is using single line POs, and does not require approval before opening, then have a setting that allows for the PO to be saved and opened all at once.
https://issues.folio.org/browse/UIOR-281
## Approach
Check  `isOpenOrderEnabled` setting, order `workflowStatus` and `approved`
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/LayerCollection/LayerPOLine.js [hunk 1]
```diff
@@ -156,18 +166,40 @@ class LayerPOLine extends Component {
...
       .then(() => {
         parentMutator.query.update({
           _path: `${pathname}`,
           layer: null,
         });
       })
       .catch(e => this.handleErrorResponse(e, line));
+
+    if (saveAndOpen) {
+      this.openOrder();
```
Comment (NikitaSedyx) [inline src/components/LayerCollection/LayerPOLine.js]: why do you need to invoke it here when it was invoked in promise chain?

File: src/components/LayerCollection/LayerPOLine.js [hunk 2]
```diff
@@ -156,18 +166,40 @@ class LayerPOLine extends Component {
     }
   };
 
-  updatePOLine = (data) => {
+  openOrder = async (saveAndOpen) => {
+    const { mutator } = this.props;
+    const order = this.getOrder();
+    const updatedOrder = await mutator.order.PUT({
```
Comment (NikitaSedyx) [inline src/components/LayerCollection/LayerPOLine.js]: are you sure about this? it sends request to update order in both cases

File: src/components/LayerCollection/LayerPOLine.js (hunk 1 above)
Comment (AndreiShumski2) [reply src/components/LayerCollection/LayerPOLine.js]: Just forgot to remove it

File: src/components/POLine/POLineForm.js [hunk 3]
```diff
@@ -149,15 +158,29 @@ class POLineForm extends Component {
...
+          disabled={pristine || submitting}
+          onClick={handleSubmit}
+        >
+          {label}
+        </Button>
+        {isSaveAndOpenButtonVisible && (
+          <Button
+            data-test-button-save-and-open
+            type="submit"
+            buttonStyle="primary mega"
```
[the remark this answers is not in the document]
Comment (AndreiShumski2) [reply src/components/POLine/POLineForm.js]: Fixed

File: src/components/LayerCollection/LayerPOLine.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/components/LayerCollection/LayerPOLine.js]: Fixed

**Pull request merged**